components/Builder.tsx

```diff
@@ -69,117 +42,59 @@ export default function Builder() {
     serialize: (formation: string[]) => btoa(formation.join(",")),
     defaultValue: new Array<string>(13).fill(""),
   });
-  const charactersNotInFormation = Characters.filter(
-    (character) => !formation.includes(character.name),
-  );
-  const [characters, setCharacters] = useState<Character[]>(
-    charactersNotInFormation.sort(),
-  );
-  const [spell, setSpell] = useQueryState<string>("spell", {
+  const [artefact, setArtefact] = useQueryState<string>("artefact", {
     parse: (query: string): string => atob(query),
-    serialize: (spell: string) => btoa(spell),
+    serialize: (artefact: string) => btoa(artefact),
     defaultValue: "blazing",
   });
   const [layout, setLayout] = useQueryState<number>("map", {
     parse: (query: string): number => parseInt(query),
     serialize: (layout: number) => layout.toString(),
     defaultValue: 0,
   });
-  const [characterFilter, setCharacterFilter] = useState<CharacterFilterType>({ class: "All", faction: "All" });
+  const [characterFilter, setCharacterFilter] = useState<CharacterFilterType>({
+    class: "All",
+    faction: "All",
+  });
+
+  const charactersNotInFormation = Object.values(Characters).filter(
+    (character) => !formation.includes(character.name),
+  );
+  const charactersInFormation = formation.map((x) => {
+    if (x === "" || x === undefined) {
+      return undefined;
+    }
+    return Characters[x.toLowerCase()];
+  });
```

Comment:
Might make more sense to chain a filter into a map here so we're not return an array with undefined valeus
```
formation.filter(x=>x).map(x=>blah)
```

Reply:
we need the empty spaces in the array to ensure characters are in the right numbered slot

Reply:
Honestly, reworking the formation as a whole to be a 2D Array would be ideal here. And everything should be ID based, not string based. But that's a bigger rework and for the future.

```
"Grid": [
                [1, 1, 1, 0],
                [1, 1, 1, 1],
                [1, 1, 1, 1],
                [0, 0, 1, 1]
            ]
```

1 is an available space, 0 is not. And then when you drop a hero into the grid, that 1 would change to being an ID of the hero, so say Reinier is 51, that entry would become 51.

Reply:
> Honestly, reworking the formation as a whole to be a 2D Array would be ideal here. And everything should be ID based, not string based. But that's a bigger rework and for the future.
> 
> ```
> "Grid": [
>                 [1, 1, 1, 0],
>                 [1, 1, 1, 1],
>                 [1, 1, 1, 1],
>                 [0, 0, 1, 1]
>             ]
> ```
> 
> 1 is an available space, 0 is not. And then when you drop a hero into the grid, that 1 would change to being an ID of the hero, so say Reinier is 51, that entry would become 51.

I’m going to make this change as a fast follow to this PR landing